rsconnect/api.py

```diff
@@ -268,6 +268,312 @@ def output_task_log(task_status, last_status, log_callback):
         return new_last_status
 
 
+class RSConnectExecutor:
+    def __init__(self, *args, **kwargs) -> None:
+        print(kwargs)
+        self.reset()
+        self._d = kwargs
+        self.setup_connect_server(*args, **kwargs)
+        self.setup_connect(*args, **kwargs)
+
+    def reset(self):
+        self._d = None
+        self.connect_server = None
+        self.connect = None
+        return self
+
+    def setup_connect_server(self, *args, **kwargs):
+        url = self.get("server", **kwargs)
+        api_key = self.get("api_key", **kwargs)
+        insecure = self.get("insecure", **kwargs) or False
+        ca_data = self.get("ca_data", **kwargs)
+        self.connect_server = RSConnectServer(url, api_key, insecure, ca_data)
+
+    def setup_connect(self, *args, **kwargs):
+        cookies = self.get("cookies", **kwargs)
+        timeout = self.get("timeout", **kwargs) or 30
+        self.connect = RSConnect(self.connect_server, cookies, timeout)
+
+    @property
+    def state(self):
+        return self._d
+
+    def get(self, *args, **kwargs):
+        key = args[0]
+        return kwargs.get(key) or self.state.get(key)
+
+    def pipe(self, func, *args, **kwargs):
+        func(*args, **kwargs)
+        return self
```

Comment:
Something feels a little off about this method. If I'm in the middle of a set of chained method calls, and I need to call an unrelated function, I feel like it's better to break the chain and call the function.

If the intent is to allow inspection of the executor state mid-stream, then the called function should probably receive `self` as an argument.